time_stepping.cpp

```diff
@@ -123,6 +123,14 @@ main(int argc, char* argv[])
         // Visualization files info.
         double viz_dump_time_interval = input_db->getDouble("VIZ_DUMP_TIME_INTERVAL");
         double next_viz_dump_time = 0.0;
+        // At specified intervals, write visualization files
+        if (IBTK::abs_equal_eps(loop_time, next_viz_dump_time, 0.1 * dt) || loop_time >= next_viz_dump_time)
+        {
+            pout << "\nWriting visualization files...\n\n";
+            ins_integrator->setupPlotData();
+            visit_data_writer->writePlotData(patch_hierarchy, iteration_num, loop_time);
+            next_viz_dump_time += viz_dump_time_interval;
+        }
         // Main time step loop
         while (!IBTK::rel_equal_eps(loop_time, time_end) && ins_integrator->stepsRemaining())
         {
```

Comment:
Btw, I should've mentioned this during the last pr, but I was running into an error on my machine when setupPlotData() was being called. The error was saying the number of arguments passed to hier_math_ops.interp was more than expected. When I made the follow modification (i.e. interpolating side centered data to cell centers) inside the setupPlotDataSpecialized() routine in INSVCTwoFluidStaggeredHierarchyIntegrator:

```
hier_math_ops.interp(un_draw_idx,  d_un_draw_var, un_idx,  d_un_sc_var,  nullptr, d_integrator_time,                        synch_cf_interface);
hier_math_ops.interp(us_draw_idx,  d_us_draw_var, us_idx,  d_us_sc_var,  nullptr, d_integrator_time,                        synch_cf_interface);
```

where d_un_draw_var and d_us_draw_var were changed to a cell variables from node variables, the error was resolved. Not sure if the initial error is due to me having a different build of IBAMR.

Reply:
You are using an out-of-date version of IBAMR. This was fixed in PR #1493 of IBAMR, which was merged November 15, 2022.